go/vt/vtgate/planbuilder/select2.go

```diff
@@ -0,0 +1,367 @@
+	PlanID PlanID
+	// Keypsace represents the keyspace to which
+	// the query will be sent.
+	Keyspace *Keyspace
+	// Vindex represents the vindex that will be used
+	// to resolve the route.
+	Vindex Vindex `json:",omitempty"`
+	// Values represents a single value or a list of
+	// values that will be used as input to the Vindex
+	// to compute the target shard(s) where the query must
+	// be sent.
+	// TODO(sougou): explain contents of Values.
+	Values interface{} `json:",omitempty"`
+}
+
+// buildSelectPlan2 is the new function to build a Select plan.
+// TODO(sougou): rename after deprecating old one.
+func buildSelectPlan2(sel *sqlparser.Select, schema *Schema) (PlanBuilder, *SymbolTable, error) {
+	return processTableExprs(sel.From, schema)
+}
+
+// processTableExprs analyzes the FROM clause. It produces a PlanBuilder
+// and the associated SymbolTable with all the routes identified.
+func processTableExprs(tableExprs sqlparser.TableExprs, schema *Schema) (PlanBuilder, *SymbolTable, error) {
+	if len(tableExprs) != 1 {
+		// TODO(sougou): better error message.
+		return nil, nil, errors.New("no list")
+	}
+	return processTableExpr(tableExprs[0], schema)
+}
+
+// processTableExpr produces a PlanBuilder subtree and SymbolTable
+// for the given TableExpr.
+func processTableExpr(tableExpr sqlparser.TableExpr, schema *Schema) (PlanBuilder, *SymbolTable, error) {
+	switch tableExpr := tableExpr.(type) {
+	case *sqlparser.AliasedTableExpr:
+		return processAliasedTable(tableExpr, schema)
+	case *sqlparser.ParenTableExpr:
+		planBuilder, symbols, err := processTableExprs(tableExpr.Exprs, schema)
+		if route, ok := planBuilder.(*RouteBuilder); ok {
+			route.From = tableExpr
+		}
+		return planBuilder, symbols, err
+	case *sqlparser.JoinTableExpr:
+		return processJoin(tableExpr, schema)
+	}
+	panic("unreachable")
+}
+
+// processAliasedTable produces a PlanBuilder subtree and SymbolTable
+// for the given AliasedTableExpr.
+func processAliasedTable(tableExpr *sqlparser.AliasedTableExpr, schema *Schema) (PlanBuilder, *SymbolTable, error) {
+	switch expr := tableExpr.Expr.(type) {
+	case *sqlparser.TableName:
+		route, table, err := getTablePlan(expr, schema)
+		if err != nil {
+			return nil, nil, err
+		}
+		planBuilder := &RouteBuilder{
+			From:  tableExpr,
+			order: 1,
+			Route: route,
+		}
+		alias := expr.Name
+		if tableExpr.As != "" {
+			alias = tableExpr.As
+		}
+		symbols := NewSymbolTable(alias, table, planBuilder)
+		return planBuilder, symbols, nil
+	case *sqlparser.Subquery:
+		// TODO(sougou): implement.
+		return nil, nil, errors.New("no subqueries")
+	}
+	panic("unreachable")
+}
+
+// getTablePlan produces the initial Route for the specified TableName.
+// It also returns the associated vschema info (*Table) so that
+// it can be used to create the symbol table entry.
+func getTablePlan(tableName *sqlparser.TableName, schema *Schema) (*Route, *Table, error) {
+	if tableName.Qualifier != "" {
```

Comment:
Is “Qualifier” a qualifier for a database? I tried to find a comment about the meaning of this field in sql.y and ast.go and didn’t. Please put one. Also explain why you’re not supporting database qualifiers (yet? At all?).